umbraco/Umbraco-CMS
# Netcore: Migrate cache refresher events to IEventAggregator pattern

### Description
- Migrated the CacheRefresher events to the `IEventAggregator` pattern.
- Introduced separate Notifications for each type.
- Changed the `ExamineComponent` to INotificationHandler<UmbracoApplicationStarting>.
- Changed the `ExamineComposer` to extension method on `IUmbracoBuilder`.

#### Bonus
- Fixed issue with previews. I think it was introduced when we change to use pure `ClaimsIdentity`. Basically we did not check if there was more than one identity on the `ClaimsPricipal`, but we have to identities if the request is authenticated for previews.
  - The new implementation is more similar to the existing
      - https://github.com/umbraco/Umbraco-CMS/pull/9833/files#diff-b23d9134f545dae5a9d148f2640f313bf05b2f04175052c3cd40dcb79feaab18


#### Areas of special interest
- Please check that the changes I had to do regarding `MainDom.Register` are correct. Basically, instead of registering the notification handlers in ``MainDom.Register`, I return from all notification handlers if we are not maindom.
   - https://github.com/umbraco/Umbraco-CMS/pull/9974/files#diff-600fe97c97149564585628f39b50d32a7a6b369729506586c1b5665d575e119a
- I removed the sender information. We used that in tests, but I really don't see how that can be interesting in a real-world example. If we still need that, we can add it as a property on the notification.


### Test
- Change previews are still updated, when saved (this is using events)
- Check that ModelsBuilder is still generating models when the Doc Type is changed. (Also using these events)
